service/db.go

```diff
@@ -634,3 +636,53 @@ func insertJWToken(token string, header *auth.Header, body *auth.Body, secret *p
 
 	return nil
 }
+
+// getExistingToken looks up existing user and grabs row where token is not expired
+// Returns tokenRow object if existing token is found and unexpired, nil if not found, else errors
+func getExistingToken(uuid string) (*tokenRow, error) {
+	if err := validation.ValidateUserUUID(uuid); err != nil {
+		return nil, authconst.ErrInvalidUUID
+	}
+
+	command := `SELECT uuid, permission, token_string, user_security.tokens.secret_key, 
+       				user_security.secret.created_timestamp, user_security.secret.expiration_timestamp
+				FROM user_security.tokens
+				INNER JOIN user_security.secret
+				ON user_security.secret.secret_key = user_security.tokens.secret_key
+				WHERE uuid = $1 AND NOW() AT TIME ZONE 'UTC' < expiration_date
+				`
+
+	row, err := postgresDB.Query(command, uuid)
+	if err != nil {
+		return nil, err
+	}
+
+	defer row.Close()
+	for row.Next() {
+		var retrievedUUID, permission, token, secret string
+		var secretCreatedTimestamp, secretExpirationTimestamp time.Time
+
+		err := row.Scan(&retrievedUUID, &permission, &token, &secret,
+			&secretCreatedTimestamp, &secretExpirationTimestamp)
+		if err != nil {
+			return nil, err
+		}
+
+		if uuid != retrievedUUID {
+			return nil, authconst.ErrInvalidUUID
+		}
+
+		return &tokenRow{
+			uuid:       retrievedUUID,
+			permission: permission,
+			token:      token,
+			secret: &pblib.Secret{
+				Key:                 secret,
+				CreatedTimestamp:    secretCreatedTimestamp.Unix(),
+				ExpirationTimestamp: secretExpirationTimestamp.Unix(),
+			},
+		}, nil
+	}
+
+	return nil, nil
```

Comment:
This needs to return an error. I understand that returning `nil` can be ideal, but I would rather return an error here.

`if err != nil` then make a new token